sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -854,3 +857,208 @@ test_suite(
+        compare-sections
+
+        echo :::: Update registers before calling functions.\\n
+        set $sp = _stack_end
+        set $gp = __global_pointer$
+        info registers
+
+        # When testing SRAM execution, we want to be sure the code is running
+        # out of SRAM and not the instruction cache.
+        echo :::: Invalidate the icache.\\n
+        print icache_invalidate()
+
+        echo :::: Call sram_main().\\n
+        print sram_main()
+
+        echo :::: Done.\\n
+"""
+
+[
+    bitstream_splice(
+        name = "rom_otp_{}_exec_disabled".format(otp_name),
+        src = "//hw/bitstream:rom",
+        data = "img_{}_exec_disabled".format(otp_name),
+        meminfo = "//hw/bitstream:otp_mmi",
+        tags = [
+            "manual",
+            "vivado",
+        ],
+        update_usr_access = True,
+        visibility = ["//visibility:private"],
+    )
+    for otp_name in OTP_IMAGES
+]
+
+[
+    opentitan_gdb_fpga_cw310_test(
+        name = "sram_program_fpga_cw310_test_otp_" + otp_name,
+        timeout = "short",
+        exit_success_pattern = "sram_program\\.c:\\d+\\] PC: 0x100020e0, SRAM: \\[0x10000000, 0x10020000\\)",
+        gdb_script = SRAM_JTAG_INJECTION_GDB_SCRIPT,
+        gdb_script_symlinks = {
+            "//sw/device/examples/sram_program:sram_program_fpga_cw310.elf": "sram_program.elf",
+        },
+        rom_bitstream = ":rom_otp_{}_exec_disabled".format(otp_name),
+        rom_kind = "Rom",
+        tags = [
+            "cw310",
+            "vivado",
+        ],
+    )
+    for otp_name in OTP_IMAGES
+]
+
+test_suite(
+    name = "rom_e2e_jtag_inject_tests",
+    tags = [
+        "cw310",
+        "vivado",
+    ],
+    tests = ["sram_program_fpga_cw310_test_otp_" + otp_name for otp_name in OTP_IMAGES],
+)
+
+[
+    opentitan_gdb_fpga_cw310_test(
+        name = "asm_interrupt_handler_fpga_cw310_test_otp_" + otp_name,
+        timeout = "short",
+        gdb_script = """
+            target extended-remote :3333
+
+            echo :::: Send OpenOCD the 'reset halt' command.\\n
+            monitor reset halt
+
+            echo :::: Load ROM symbols into GDB.\\n
+            file rom.elf
+
+            echo :::: Set breakpoint on exception handler.\\n
+            break _asm_exception_handler
+
+            echo :::: Attempt to trigger an exception.\\n
+            set $pc = 0
+            continue
+            printf ":::: PC=%p. Expected PC=%p.\\n", $pc, _asm_exception_handler
+
+            if $pc == _asm_exception_handler
+                echo :::: Test passed!\\n
+                quit 0
+            else
+                echo :::: Test failed!\\n
+                quit 1
+            end
+        """,
+        gdb_script_symlinks = {
+            "//sw/device/silicon_creator/rom:rom_fpga_cw310.elf": "rom.elf",
+        },
+        rom_bitstream = ":rom_otp_{}_exec_disabled".format(otp_name),
+        rom_kind = "Rom",
+        tags = [
+            "cw310",
+            "vivado",
```

Comment:
TODO(dmcardle) Reevaluate the tags once we're happy with #15713.

Reply:
Actually, I'd prefer to merge this before #15713. This PR makes it easier to deal with optional test coordinator flags.